src/python/zquantum/core/wip/circuits/_compatibility.py

```diff
@@ -9,4 +9,9 @@
 
 
 def new_circuit_from_old_circuit(old: OldCircuit) -> NewCircuit:
-    return import_from_cirq(old.to_cirq())
+    if not old.qubits:
+        return NewCircuit([])
+
+    pre_new_circuit = import_from_cirq(old.to_cirq())
+    n_qubits = max([qubit.index for qubit in old.qubits]) + 1
+    return NewCircuit(pre_new_circuit.operations, n_qubits=n_qubits)
```

Comment:
Will it work as expected if `old` contains any custom gates? The former `import_from_cirq(old.to_cirq())` one-liner would capture that, assuming we support importing custom gates from Cirq

Reply:
Ahh, nevermind. I forgot we don't pass custom gate definitions via `__init__` anymore.